src/widget.ts

```diff
@@ -104,14 +104,33 @@ export class DeephavenView extends DOMWidgetView {
     }
   };
 
+  //@ts-ignore
+  onDisconnect = (type): void => {
+    log.info(`Kernel ${type}, removing iframe`);
+    this.iframe.remove();
+  }
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   render(): any {
     const iframeUrl = this.model.get('iframe_url');
     const width = this.model.get('width');
     const height = this.model.get('height');
+    let removed = false;
 
     window.addEventListener('message', this.handleAuthentication);
 
+    //@ts-ignore
+    const context = this.model.widget_manager._context;
+
+    this.model.on('change:kernel_active', () => this.onDisconnect("exiting"), this)
+
+    context.sessionContext.statusChanged.connect((sender: any, args: string) => {
```

Comment:
I'd add a comment right here to where you're pulling this from, since this doesn't seem to be well documented.

Reply:
Also - which listeners are you concerned about being spawned multiple times? For this listener, can we add it in the contructor like it's added in manager.ts? https://github.com/jupyter-widgets/ipywidgets/blob/main/python/jupyterlab_widgets/src/manager.ts#L427